sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/RateSourceProvider.scala

```diff
@@ -199,13 +199,52 @@ class RateStreamSource(
     }
 
     val localStartTimeMs = startTimeMs + TimeUnit.SECONDS.toMillis(startSeconds)
-    val relativeMsPerValue =
-      TimeUnit.SECONDS.toMillis(endSeconds - startSeconds) / (rangeEnd - rangeStart)
```

Comment:
I thought that you would only change `TimeUnit.SECONDS.toMillis(endSeconds - startSeconds).toDouble`. Wasn't expecting all this change!